packages/@n8n/ai-workflow-builder.ee/evaluations/programmatic/evaluators/from-ai.ts

```diff
@@ -0,0 +1,96 @@
+import type { INodeTypeDescription } from 'n8n-workflow';
+
+import type { SimpleWorkflow } from '@/types';
+
+import type { Violation } from '../../types/evaluation';
+import type { SingleEvaluatorResult } from '../../types/test-result';
+import { isTool } from '../../utils/is-tool';
+import { calcSingleEvaluatorScore } from '../../utils/score';
+
+/**
+ * Checks if a value contains $fromAI or $fromAi reference
+ */
+function containsFromAi(value: unknown): boolean {
+	if (typeof value !== 'string') {
+		return false;
+	}
+
+	// Check for $fromAI or $fromAi patterns (case-insensitive variations)
+	return /\$from[Aa][Ii]\(.+\)/.test(value);
```

Comment:
A simpler way could be to just use case-insensitive flag
```suggestion
	return /\$fromai\(.+\)/i.test(value);
```